charms/slurmctld/src/charm.py

```diff
@@ -188,9 +223,46 @@ def _on_slurmdbd_available(self, event: SlurmdbdAvailableEvent) -> None:
         self._stored.slurmdbd_host = event.slurmdbd_host
         self._on_write_slurm_conf(event)
 
-    def _on_slurmdbd_unavailable(self, _: SlurmdbdUnavailableEvent) -> None:
+    def _on_slurmdbd_unavailable(self, event: SlurmdbdUnavailableEvent) -> None:
         self._stored.slurmdbd_host = ""
-        self._check_status()
+        self._on_write_slurm_conf(event)
+
+    def _on_influxdb_available(self, event: InfluxDBAvailableEvent) -> None:
+        """Assemble the influxdb acct_gather.conf options."""
+        logger.debug("## Generating acct gather configuration")
+
+        acct_gather_params = {
+            "ProfileInfluxDBDefault": ["ALL"],
+            "ProfileInfluxDBHost": event.influxdb_host,
+            "ProfileInfluxDBUser": event.influxdb_user,
+            "ProfileInfluxDBPass": event.influxdb_pass,
+            "ProfileInfluxDBDatabase": event.influxdb_database,
+            "ProfileInfluxDBRTPolicy": event.influxdb_policy,
+        }
+        acct_gather_params.update({"SysfsInterfaces": interfaces()})
+        logger.debug(f"## _stored.acct_gather_params: {acct_gather_params}")
```

Comment:
Do we want this information logged here? It will reveal the InfluxDB password to anyone that has access to the juju log stream.